app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,38 @@
+class MerchantsController < ApplicationController
+
+  def index
+    @merchants = Merchant.all
+  end
+
+  def new
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def edit
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    @merchant = Merchant.find(params[:id])
+    @merchant.update(merchant_params)
+    redirect_to "/merchants/#{@merchant.id}"
+  end
+
+  def destroy
+    Merchant.destroy(params[:id])
+    redirect_to '/merchants'
+  end
```

Comment:
I haven't looked at your tests yet, but this will only work if your merchant does not have any items when you're testing. There will be an error that happens when you try to delete a merchant that has items. It will be a dependency error since some items belong to that merchant. You'd have to somehow find all of the items that belong to that merchant, delete those first, and then destroy the merchant.